### PR TITLE
Vigenere Cipher in 'C' language

### PR DESCRIPTION
**Pull request by faisal1337**
Vigenere Cipher implemented in 'C' language.

### CODE REVIEW

**Comment by faisal1337** [comment]
@AdiChat

File: code/cryptography/ciphers/caesar.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include <cs50.h>
```
Comment (AdiChat) [inline code/cryptography/ciphers/caesar.c]: Kindly include the header file `cs50.h` 👍 

It will be good to merge then 👍

**Comment by faisal1337** [comment]
Check it out now.

File: code/cryptography/ciphers/cs50.c [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+ Watch 20
+  Star 85  Fork 45 cs50/libcs50
+ Code  Issues 2  Pull requests 1 Insights 
+Branch: develop Find file Copy pathlibcs50/src/cs50.c
+80275e4  26 days ago
+@crossroads1112 crossroads1112 added get_long (#109)
+3 contributors @dmalan @crossroads1112 @kzidane
+RawBlameHistory     
+689 lines (615 sloc)  18.2 KB
+/**
```
Comment (AdiChat) [inline code/cryptography/ciphers/cs50.c]: Lines 1 to 19 is not part of code. Kindly rectify 👍

File: code/cryptography/ciphers/cs50.c [hunk 3]
```diff
@@ -0,0 +1,718 @@
+Skip to content
```
Comment (AdiChat) [inline code/cryptography/ciphers/cs50.c]: This file is not being used. Kindly rectify 👍

File: code/cryptography/ciphers/cs50.c [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#define _GNU_SOURCE
```
Comment (AdiChat) [inline code/cryptography/ciphers/cs50.c]: You might need to write your own version 👍 
above comment might be removed 👍

File: code/cryptography/ciphers/cs50.h [hunk 5]
```diff
@@ -0,0 +1,252 @@
+Skip to content
```
Comment (AdiChat) [inline code/cryptography/ciphers/cs50.h]: Same errors persist. Kindly rectify 👍

File: code/cryptography/ciphers/cs50.h [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+ * inputs only a line ending, returns "", not NULL. Returns NULL
+ * upon error or no input whatsoever (i.e., just EOF). Stores string
+ * on heap, but library's destructor frees memory on program's exit.
+ */
+string get_string(va_list *args, const string format, ...) __attribute__((format(printf, 2, 3)));
+string GetString(void) __attribute__((deprecated));
+#define get_string(...) __extension__ IF_ELSE(ISEMPTY(__VA_ARGS__))(WARN(get_string(NULL, NULL)))(get_string(NULL, __VA_ARGS__))
+
+#endif
+� 2017 GitHub, Inc.
```
Comment (AdiChat) [inline code/cryptography/ciphers/cs50.h]: The following lines are certainly not part of any code. Kindly rectify 👍

**Comment by AdiChat** [comment]
You may submit a new pull request with modified code 👍 
See [C++ code](https://github.com/OpenGenus/cosmos/blob/master/code/cryptography/ciphers/vigenere_cypher.cpp) for reference 👍

**Comment by arnavb** [comment]
@faisal1337 For a simple Vigenere cipher, you don't really need a library. Please simplify! @AdiChat please close!

**Comment by AdiChat** [comment]
@faisal1337 We have closed this pull request but you can always open another pull request for an addition you wish to make. :+1:

**Pull request closed without merging**